File: hw/chicaiza/u2/hw19Inheritance/FarmSystemPython/model/sheep.py
from datetime import date
from model.farm_animal import FarmAnimal

class Sheep(FarmAnimal):
    def __init__(self, last_sheering: date, id, breed, born_on, gender,
                 is_able_to_reproduce, weight, cage):
        super().__init__(id, breed, born_on, gender, is_able_to_reproduce, weight, cage)
        self.last_sheering = last_sheering

    def cut_wool(self):
        self.last_sheering = date.today()

    def __str__(self):
        return f"Sheep(lastSheering={self.last_sheering}, {super().__str__()})"

File: ws/gudino/u2/ws12Inheritance/Farmsystem/src/main/java/ec/edu/espe/farmsymulator/model/Sheep.java
package ec.edu.espe.farmsymulator.model;

import java.util.Date;

/**
 *
 * @author Bryan Gudino
 */
public class Sheep extends FarmAnimal {
    private Date lastSheering;

    public Sheep(Date lastSheering, int id, String breed, Date bornOn, String gender, boolean isAbleToReproduce, float weight, Cage cage) {
        super(id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
        this.lastSheering = lastSheering;
    }

    @Override
    public String toString() {
        return "Sheep{" + "lastSheering=" + lastSheering + "," + super.toString() + '}';
    }

    public void shear() {
        this.lastSheering = new Date();
    }
    
    public void cutWhool() {
        System.out.println("Cutting the wool of the sheep...");
    }

    /**
     * @return the lastSheering
     */
    public Date getLastSheering() {
        return lastSheering;
    }

    /**
     * @param lastSheering the lastSheering to set
     */
    public void setLastSheering(Date lastSheering) {
        this.lastSheering = lastSheering;
    }
}